### PR TITLE
Incorrect height of vehicle restriction signs

### LINKED ISSUE #741: Vehicle restrictions overlay icons appear at base of hydro dam
This issue seems specific to vehicle restricitons tool (found during testing of PR #721):

image

The hydro dam in that image is workshop asset: [Medium Road Dam](https://steamcommunity.com/sharedfiles/filedetails/?id=1939679056)

### PR DESCRIPTION
**Pull request by krzychu124**
Closes #741 

- fixes incorrect height of drawn vehicle restriction signs - segments on top of the dam have huge bounding box. 
- force overlay height to be at least at terrain level - improves usability

### CODE REVIEW

**Review by chameleon-tbn — APPROVED** [review]
lgtm now

File: TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs [hunk 1]
```diff
@@ -450,7 +450,12 @@ public VehicleRestrictionsTool(TrafficManagerTool mainTool)
                 return false;
             }
 
-            Vector3 center = segment.m_bounds.center;
+            Vector3 center = segment.m_middlePosition;
+            float terrainY = TerrainManager.instance.SampleDetailHeightSmooth(center);
+            if (terrainY > center.y) {
```
Comment (kianzarrin) [inline TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs:455]: I don't understand what is this line for?
this is how it looks like 
image
and this is how it looks like when I comment out the if statement.
image
Isn't the second one better?

Comment (kianzarrin) [reply TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs:455]: If anything I think that orange overlay should come down.

What do you mean by sunken anyway? you mean like this(the if statement makes no difference in here)?
image

Comment (krzychu124) [reply TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs:455]: Try vehicle restrictions on the dam 😉

Comment (kianzarrin) [reply TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs:455]: but don't you think in trying to fix the dam you might have made the normal use case worse?

Comment (kianzarrin) [reply TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs:455]: the dam seems fine with this condition commented out.
image
So I still don't understand what this if condition does or how sunken segment is relevant to dam.

**Review by DaEgi01 — APPROVED** [review]

File: TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs (hunk 1 above)
Comment (krzychu124) [reply TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs]: Changed 😉

Comment (kianzarrin) [reply TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs]: cool it makes more sense now :)

**Review by kianzarrin — APPROVED** [review]

**Pull request merged**